core/schema/git.go

```diff
@@ -123,191 +156,202 @@ func (s *gitSchema) git(ctx context.Context, parent dagql.Instance[*core.Query],
 		if err != nil {
 			return inst, err
 		}
-		svcs = append(svcs, core.ServiceBinding{
+		gitServices = append(gitServices, core.ServiceBinding{
 			ID:       svc.ID(),
 			Service:  svc.Self,
 			Hostname: host,
 		})
 	}
 
-	// 2. Get client metadata
-	clientMetadata, err := engine.ClientMetadataFromContext(ctx)
-	if err != nil {
-		return inst, fmt.Errorf("failed to get client metadata from context: %w", err)
-	}
+	var (
+		sshAuthSock    dagql.Instance[*core.Socket]
+		httpAuthToken  dagql.Instance[*core.Secret]
+		httpAuthHeader dagql.Instance[*core.Secret]
+	)
 
-	// 3. Setup authentication
-	var authSock dagql.Instance[*core.Socket]
-	var authToken dagql.Instance[*core.Secret]
-
-	// First parse the ref scheme to determine auth strategy
-	remote, err := gitutil.ParseURL(args.URL)
-	if errors.Is(err, gitutil.ErrUnknownProtocol) {
-		remote, err = gitutil.ParseURL("https://" + args.URL)
-	}
-	if err != nil {
-		return inst, fmt.Errorf("failed to parse Git URL: %w", err)
-	}
-	if remote.Scheme == "ssh" && (remote.User == nil || remote.User.Username() == "") {
-		// default to git user for SSH, otherwise weird incorrect defaults like "root" can get
-		// applied in various places. This matches the git module source implementation.
+	switch remote.Scheme {
+	case gitutil.SSHProtocol:
 		if remote.User == nil {
+			// default to git user for SSH, otherwise weird incorrect defaults
+			// like "root" can get applied in various places. This matches the
+			// git module source implementation.
 			remote.User = url.User("git")
 		}
-		pass, ok := remote.User.Password()
-		if ok {
-			remote.User = url.UserPassword(remote.User.Username(), pass)
-		}
```

Comment:
@sipsma so this was part of a git fix you did in [`68b2782` (#9682)](https://github.com/dagger/dagger/pull/9682/commits/68b2782cc3dd5ca2bf7a08a8ea34b01d22e3b35f)

But, I'll be honest, I can't quite tell what this bit is for? It seems to just add the password... which is already in there? Just wanted to check.

This whole URL manipulation actually gets so much neater with the forked `gitutil` package, since now there's `remote.String()`, which allows us to manipulate any of those URL fields inline.

Reply:
Believe there was an intention here to handle a case of `username == "" && userpassword != ""`, but it's off obviously. I was in the midst of trying to make all the pre-existing git integ tests pass when I added this code so I probably just moved on after they passed and didn't notice this bit was no-op. So nothing significant, thanks for cleaning up